### PR TITLE
Updates for Node version 4.x

### PR DESCRIPTION
**Pull request by dfbaskin**
There is still more to do with the source, so I don't expect this to be merged, but I wanted to get the conversation started.

The things that are still remaining are:

- Check compilation for other preprocessor definitions (DEBUG, UNICODE, dynodbc)
- Verify that it builds on OSX and Linux (I performed the update on a Windows machine)
- Verify the tests

Right now, it builds correctly with the default settings, but most of the tests are failing, but I don't have the all of the test databases and ODBC connections configured correctly yet.

### CODE REVIEW

File: src/odbc.cpp [hunk 1]
```diff
@@ -496,10 +496,10 @@ Handle<Value> ODBC::GetColumnValue( SQLHSTMT hStmt, Column column,
         //at the specified time.
         timeInfo.tm_isdst = -1;
 #ifdef TIMEGM
-        return NanEscapeScope(NanNew<Date>((double(timegm(&timeInfo)) * 1000)
+        return scope.Escape(Nan::New<Date>((double(timegm(&timeInfo)) * 1000)
                           + (odbcTime.fraction / 1000000)));
```
Comment (wankdanker) [inline src/odbc.cpp]: + (odbcTime.fraction / 1000000)).ToLocalChecked());

File: src/odbc.cpp [hunk 2]
```diff
@@ -496,10 +496,10 @@ Handle<Value> ODBC::GetColumnValue( SQLHSTMT hStmt, Column column,
...
 #else
-        return NanEscapeScope(NanNew<Date>((double(timelocal(&timeInfo)) * 1000)
+        return scope.Escape(Nan::New<Date>((double(timelocal(&timeInfo)) * 1000)
                           + (odbcTime.fraction / 1000000)));
```
Comment (wankdanker) [inline src/odbc.cpp]: + (odbcTime.fraction / 1000000)).ToLocalChecked());

File: package.json [hunk 3]
```diff
@@ -25,15 +25,15 @@
     "lib": "."
   },
   "engines": {
-    "node": ">=0.8.0 <4.0.0"
+    "node": ">=4.0.0"
```
Comment (wankdanker) [inline package.json]: This can go back to just >=0.8.0 now, since NAN supports 0.8, 0.10, 0.12, +

**Comment by wankdanker** [comment]
This looks GREAT! 

I made the changes I mentioned above and it compiled and most tests succeeded on my Linux box with v4.0.0. Even though NAN is supposed to support v0.8, 0.10, 0.12, none of those compile for me. Just a few errors thrown though. Should be able to figure it out.

**Comment by wankdanker** [comment]
Here are some fixes I made that were working but now they are not: b15cbea9f787892adb761f1d6eb75d5b487a56c0. Have to run, but wanted to get these out there.

**Comment by wankdanker** [comment]
Correction, this branch works with 0.12.7 and 4.0.0: https://github.com/wankdanker/node-odbc/tree/fix-for-node-v0.12.7

**Comment by wankdanker** [comment]
This branch contains the remaining fixes for node 0.10.40 and 0.8.28: https://github.com/wankdanker/node-odbc/tree/fix-for-node-v0.10.40

**Comment by dfbaskin** [comment]
Thanks! I'll integrate those changes into this pull request.  What should we change the version number of the package to? 1.1.9? 1.2.0? 2.0.0?

**Comment by wankdanker** [comment]
I'll take care of the version bump. It's going to be 1.2.0 though. Working on merging and releasing now.

**Comment by dfbaskin** [comment]
Great, thanks!

**Comment by wankdanker** [comment]
Published odbc@1.2.0 to npm.

**Pull request merged**